app/src/main/java/com/bunbeauty/papakarlo/feature/menu/MenuFragment.kt

```diff
@@ -65,46 +77,68 @@ class MenuFragment : BaseFragment(R.layout.fragment_menu) {
 
         viewModel.getMenu()
         viewBinding.fragmentMenuCvMain.setContentWithTheme {
-            val menuState by viewModel.menuState.collectAsState()
-            MenuScreen(menuState = menuState)
+            val menuState by viewModel.menuState.collectAsStateWithLifecycle()
+            MenuScreen(menuUi = menuUiStateMapper.map(menuState))
+            LaunchedEffect(menuState.eventList) {
+                handleEventList(menuState.eventList)
+            }
         }
-        handleActions()
     }
 
-    private fun handleActions() {
-        viewModel.actionFlow.startedLaunch { action ->
-            when (action) {
-                is MenuAction.GoToSelectedItem -> {
+    private fun handleEventList(eventList: List<MenuState.Event>) {
+        eventList.forEach { event ->
+            when (event) {
+                is MenuState.Event.GoToSelectedItem -> {
                     findNavController().navigate(
                         MenuFragmentDirections.toProductFragment(
-                            action.uuid,
-                            action.name
+                            event.uuid,
+                            event.name
                         )
                     )
                 }
             }
         }
+        viewModel.consumeEventList(eventList)
     }
 
     @Composable
-    private fun MenuScreen(menuState: State<MenuUI>) {
-        when (menuState) {
-            is State.Success -> {
-                MenuSuccessScreen(menuState.data)
-            }
-            is State.Error -> {
-                ErrorScreen(R.string.error_menu_loading) {
-                    viewModel.getMenu()
+    private fun MenuScreen(menuUi: MenuUi) {
+        FoodDeliveryToolbarScreen(
+            title = stringResource(R.string.title_menu),
+            drawableId = R.drawable.logo_top_papa_karlo,
+            topActions = listOf(
+                FoodDeliveryCartAction(
+                    topCartUi = menuUi.topCartUi,
+                ) {
+                    val backQueue = findNavController().backQueue
```

Comment:
We don't need this logic here. Just `findNavController().navigate(ProductDetailsFragmentDirections.globalConsumerCartFragment())`